### DISCUSSION TITLE
Vite-plugin-ssr build mode

### DISCUSSION THREAD

**Discussion by alivadjid** [vikejs/vike#660] [Discussion]
First, thanks @brillout for your great package! It very useful :+1: 

I'm using vite-ssr-plugin in a few projects, and I found an interesting thing. 

[In this project](https://github.com/alivadjid/vite-plugin-ssr-i18n-vue3) i used vite-ssr with vue3 and custom internationalization code.

And the thing is: on development mode every works fine, user choose locale and see that text is changing. Url like this: `/fr-fr/about`. Locale Franch and page about.

But on prod mode (after start build command). User choose locale and url is changing `/fr-fr/about`. But here is appearing a message: `404 Page not found`.

I solved this error by [adding folders with the name of locale](https://github.com/alivadjid/vite-plugin-ssr-i18n-vue3/tree/master/pages). `/pages/fr-FR/[site-pages]` and adding empty vue components. And it works correct.
![изображение](https://user-images.githubusercontent.com/52418132/220203094-edbb2acf-20aa-4473-97fd-fb6a0c452329.png)

Is this a correct solution to solve this? Or maybe there is a better solution?

**Comment by brillout** [comment]
Actually, the docs about i18n with pre-rendering improved quite a lot the last couple of days: https://vite-plugin-ssr.com/i18n#pre-rendering. Let me know whether these new docs help or not.

Also make sure to use the latest VPS version (a lot of improved warnings around i18n and pre-rendering). I'll be releasing a new version tomorrow that will further improve the i18n warnings.

**Reply by brillout** [reply]
> I'll be releasing a new version tomorrow that will further improve the i18n warnings

Done.

**Reply by alivadjid** [reply]
Cool! :+1: 
What I got after update VPS.
1. Tried to use [https://vite-plugin-ssr.com/i18n#pre-rendering](https://vite-plugin-ssr.com/i18n#pre-rendering).

`vite.config.ts`
```javascript
plugins: [
    vue(),
    ssr({ prerender: true, includeAssetsImportedByServer: true }),
  ],
```
in `_default.page.server.ts`
```javascript
export { onBeforePrerender };

function onBeforePrerender() {
  console.log("onBeforePrerender");
}
````
I don't see this console.

Is `onBeforePrerender` nessassy if locale takes from `_default.page.route.ts`, and urlOriginal change here. urlOriginal shows link without locale. For example if link `/de-de/about/` urlOriginal is `/about/`. The same how it described [https://vite-plugin-ssr.com/i18n#pre-rendering](https://vite-plugin-ssr.com/i18n#pre-rendering)
```javascript

async function onBeforeRoute(pageContext: { urlOriginal: string }) {
  let urlMod = pageContext.urlOriginal;
  const { urlWithoutLocale, locale } = await extractLocale(urlMod);
  urlMod = urlWithoutLocale;
  return {
    pageContext: {
      locale,
      urlOriginal: urlMod,
    },
  };
}
```
<hr>

2. After update and run build, warnings appeared for pages `index.page.vue`
![изображение](https://user-images.githubusercontent.com/52418132/221038915-93b0396a-da8c-49dc-9d40-ca18c37ddd03.png)
if remove this locale folder and all components.
![изображение](https://user-images.githubusercontent.com/52418132/221039481-87c9a2c9-215b-4def-b215-fbf2566d8fa8.png)
on prod and dev mode it work fine
![изображение](https://user-images.githubusercontent.com/52418132/221041411-b4b7f434-c404-44f0-bc1d-72e4008b2939.png)
But on build
![изображение](https://user-images.githubusercontent.com/52418132/221041468-06dbd547-e35d-46b9-8b4b-bc018f3a65a8.png)

**Reply by brillout** [reply]
Seems like a user land issue. But let me know if you believe it's a problem on VPS's side.
